examples/crowdfunding/assets/crowdFund.teal

```diff
@@ -0,0 +1,469 @@
+int 0
+txn ApplicationID
+==
+
+// if not creation skip this section
+bz not_creation
+
+// save the creator address to the global state
+byte "Creator"
+txn Sender
+app_global_put
+
+// verify that 5 arguments are passed
+txn NumAppArgs
+int 5
+==
+bz failed
+
+//store the start date
+byte "StartDate"
+txna ApplicationArgs 0
+btoi
+app_global_put
+
+// store the end date
+byte "EndDate"
+txna ApplicationArgs 1
+btoi
+app_global_put
+
+// store fund goal
+byte "Goal"
+txna ApplicationArgs 2
+btoi
+app_global_put
+
+// store the fund receiver
+byte "Receiver"
+txna ApplicationArgs 3
+app_global_put
+
+// set the total raised to 0 and store it
+byte "Total"
+int 0
+app_global_put
+
+// store the fund close date
+byte "FundCloseDate"
+txna ApplicationArgs 4
+btoi
+app_global_put
+
+// return a success
+int 1
+return
+
+not_creation:
+// check if this is deletion transaction
+int DeleteApplication
+txn OnCompletion
+==
+bz not_deletion
+
+// To delete the app
+// The creator must empty the escrow
+// to the fund receiver if the escrow has funds
+// only the creator
+// can delete the app
+byte "Creator"
+app_global_get
+txn Sender
+==
+
+// check that we are past fund close date
+// global LatestTimestamp
+// byte "FundCloseDate"
+// app_global_get
+// >=
+// &&
+bz failed
+
+// if escrow balance is zero
+// let the app be deleted
+// escrow account must be passed
+// into the call as an argument
+int 0
+int 1
+balance
+==
+
+// if the balance is 0 allow the delete
+bnz finished
+
+// if the escrow is not empty then 
+// there must be need two transactions
+// in a group
+global GroupSize
+int 2
+==
+
+// second tx is an payment
+gtxn 1 TypeEnum
+int 1
+==
+&&
+
+// the second payment transaction should be
+// a close out transaction to receiver
+byte "Receiver"
+app_global_get
+gtxn 1 CloseRemainderTo
+==
+&&
+
+// the amount of the payment transaction 
+// should be 0
+gtxn 1 Amount
+int 0
+==
+&&
+
+// the sender of the payment transaction
+// should be the escrow account
+byte "Escrow"
+app_global_get
+gtxn 1 Sender
+==
+&&
+bz failed
+int 1
+return
+
+not_deletion:
+// check if this is update
+int UpdateApplication
+txn OnCompletion
+==
+
+bz not_update
+// verify that the creator is
+// making the call
+byte "Creator"
+app_global_get
+txn Sender
+==
+
+// the call should pass the escrow 
+// address
+txn NumAppArgs
+int 1
+==
+&&
+bz failed
+
+// store the address in global state
+// this parameter should be addr:
+byte "Escrow"
+txna ApplicationArgs 0
+app_global_put
+int 1
+return
+
+not_update:
+// check for closeout
+int CloseOut
+txn OnCompletion
+==
+bnz close_out
+
+// check if no params are
+// passed in, which should
+// only happen when someone just
+// wants to optin
+// note the code is written
+// to allow opting in and donating with
+// one call
+int 0
+txn NumAppArgs
+==
+bz check_parms
+
+// Verify someone is 
+// not just opting in
+int OptIn
+txn OnCompletion
+==
+bz failed
+int 1
+return
+
+check_parms:
```

Comment:
Shouldn't this be `check_params`?

Reply:
https://github.com/algorand/smart-contracts/blob/master/devrel/crowdfunding/crowd_fund.teal
 
 It can be.